alf/utils/common.py

```diff
@@ -925,20 +894,33 @@ def decorate(f):
     return decorate
 
 
-def set_random_seed(seed, eager_mode):
-    """Set a seed for deterministic behaviors."""
-    if seed is not None:
-        if not tf.config.list_physical_devices('GPU'):
-            assert eager_mode, (
-                "CPU mode: because you have specified a fixed " +
-                "random seed, make sure to turn on eager mode " +
-                "to have deterministic results!")
+def set_random_seed(seed):
+    """Set a seed for deterministic behaviors.
 
-        os.environ["TF_DETERMINISTIC_OPS"] = str(1)
-    else:
+    Args:
+        seed (int|None): seed to be used. If None, a default seed based on
+            pid and time will be used.
+    Returns:
+        The seed being used if `seed` is None. Note that you may not reproduce
+        a previous experiment even if use use its seed because when seed is None,
+        cudnn.deterministic is not set to True
```

Comment:
I guess the real reason is that we cannot directly set the seed for cudnn given the previous seed?

Reply:
And the comment should be more precise: when someone runs an experiment with a pre-selected manual seed, he can definitely reproduce the results; however, if he runs the experiment with seed=None and later recovers the seed from the tensorboard, and if cudnn is used in the code, then there is no guarantee that the results will be reproduced with the recovered seed.

Reply:
updated